### DISCUSSION TITLE
Setting schedule on TRV zigbee model TS0601 manufacturer _TZE200_i48qyn9s /  _TZE200_uiyqstza

### DISCUSSION THREAD

**Discussion by louisnichols** [Koenkk/zigbee2mqtt#17605] [Devices]
Hi!

Complete noob here. :)

So I have 4 TRVs as mentioned in the title: Zigbee Model TS0601 manufacturer _TZE200_i48qyn9s /  _TZE200_uiyqstza. The label on the packaging says "Essentials model ESS-HK-TRV-6102" from the company WeSmartify.

As far as I can tell, they are identical to this one from Lidl: https://www.zigbee2mqtt.io/devices/368308_2010.html .

To get them to work, I had to configure them using a dedicated .js file, which I found online. Unfortunately, I can't quite find anymore where I downloaded this script anymore. All I can tell is that I just copy/pasted it (with the necessary changes) to get both manufacturers to work.

So that's the background, now here's the rub.

I would like to remotely set their schedules and am not able to. On their "Exposes" pages I can edit the values but the "Apply" button is disabled.

Also, I tried the instructions from here https://www.zigbee2mqtt.io/devices/368308_2010.html#monday-composite to set them via MQTT, but that also didn't work.

Specifically, I sent a message to the topic `zigbee2mqtt/Thermostat_Office/set` with the content
`{
  "monday": {
    "monday_temp_1": "14",
    "monday_hour_1": "00",
    "monday_minute_1": "00",
    "monday_temp_2": "15",
    "monday_hour_2": "02",
    "monday_minute_2": "00",
    "monday_temp_3": "16",
    "monday_hour_3": "03",
    "monday_minute_3": "00",
    "monday_temp_4": "17",
    "monday_hour_4": "04",
    "monday_minute_4": "00",
    "monday_temp_5": "18",
    "monday_hour_5": "05",
    "monday_minute_5": "00",
    "monday_temp_6": "19",
    "monday_hour_6": "06",
    "monday_minute_6": "00",
    "monday_temp_7": "20",
    "monday_hour_7": "07",
    "monday_minute_7": "00",
    "monday_temp_8": "21",
    "monday_hour_8": "08",
    "monday_minute_8": "00",
    "monday_temp_9": "22",
    "monday_hour_9": "09",
    "monday_minute_9": "00"
  }
}

`

But not much happened. When I loook at the topic zigbee2mqtt/Thermostat_Office I can see that it took my values, but that's it. They don't show on the Exposes of the device, or on the dashboard.
` 
  "away_mode": "OFF",
  "away_preset_day": 1,
  "away_preset_days": 0,
  "away_preset_hour": 0,
  "away_preset_minute": 0,
  "away_preset_month": 1,
  "away_preset_temperature": "17.0",
  "away_preset_year": 19,
  "binary_one": null,
  "binary_two": null,
  "child_lock": "UNLOCK",
  "comfort_temperature": "21.0",
  "current_heating_setpoint": "0.0",
  "current_heating_setpoint_auto": "21.0",
  "detectwindow_temperature": "15.0",
  "detectwindow_timeminute": 0,
  "eco_temperature": "16.5",
  "friday_hour_1": "06",
  "friday_hour_2": "09",
  "friday_hour_3": "17",
  "friday_hour_4": "23",
  "friday_hour_5": "24",
  "friday_hour_6": "24",
  "friday_hour_7": "24",
  "friday_hour_8": "24",
  "friday_minute_1": "00",
  "friday_minute_2": "00",
  "friday_minute_3": "00",
  "friday_minute_4": "00",
  "friday_minute_5": "00",
  "friday_minute_6": "00",
  "friday_minute_7": "00",
  "friday_minute_8": "00",
  "friday_temp_1": "17.0",
  "friday_temp_2": "21.0",
  "friday_temp_3": "17.0",
  "friday_temp_4": "21.0",
  "friday_temp_5": "17.0",
  "friday_temp_6": "21.0",
  "friday_temp_7": "17.0",
  "friday_temp_8": "21.0",
  "friday_temp_9": "17.0",
  "linkquality": 27,
  "local_temperature": "22.5",
  "local_temperature_calibration": "2.5",
`
  **"monday_hour_1": "00",
  "monday_hour_2": "02",
  "monday_hour_3": "03",
  "monday_hour_4": "04",
  "monday_hour_5": "05",
  "monday_hour_6": "06",
  "monday_hour_7": "07",
  "monday_hour_8": "08",
  "monday_minute_1": "15",
  "monday_minute_2": "00",
  "monday_minute_3": "00",
  "monday_minute_4": "00",
  "monday_minute_5": "00",
  "monday_minute_6": "00",
  "monday_minute_7": "00",
  "monday_minute_8": "00",
  "monday_temp_1": "14.0",
  "monday_temp_2": "15.0",
  "monday_temp_3": "16.0",
  "monday_temp_4": "17.0",
  "monday_temp_5": "18.0",
  "monday_temp_6": "19.0",
  "monday_temp_7": "20.0",
  "monday_temp_8": "21.0",
  "monday_temp_9": "22.0",**
  `"preset": "schedule",
  "saturday_hour_1": "06",
  "saturday_hour_2": "09",
  "saturday_hour_3": "17",
  "saturday_hour_4": "23",
  "saturday_hour_5": "24",
  "saturday_hour_6": "24",
  "saturday_hour_7": "24",
  "saturday_hour_8": "24",
  "saturday_minute_1": "00",
  "saturday_minute_2": "00",
  "saturday_minute_3": "00",
  "saturday_minute_4": "00",
  "saturday_minute_5": "00",
  "saturday_minute_6": "00",
  "saturday_minute_7": "00",
  "saturday_minute_8": "00",
  "saturday_temp_1": "17.0",
  "saturday_temp_2": "21.0",
  "saturday_temp_3": "17.0",
  "saturday_temp_4": "21.0",
  "saturday_temp_5": "17.0",
  "saturday_temp_6": "21.0",
  "saturday_temp_7": "17.0",
  "saturday_temp_8": "21.0",
  "saturday_temp_9": "17.0",
  "sunday_hour_1": "06",
  "sunday_hour_2": "09",
  "sunday_hour_3": "17",
  "sunday_hour_4": "23",
  "sunday_hour_5": "24",
  "sunday_hour_6": "24",
  "sunday_hour_7": "24",
  "sunday_hour_8": "24",
  "sunday_minute_1": "00",
  "sunday_minute_2": "00",
  "sunday_minute_3": "00",
  "sunday_minute_4": "00",
  "sunday_minute_5": "00",
  "sunday_minute_6": "00",
  "sunday_minute_7": "00",
  "sunday_minute_8": "00",
  "sunday_temp_1": "17.0",
  "sunday_temp_2": "21.0",
  "sunday_temp_3": "17.0",
  "sunday_temp_4": "21.0",
  "sunday_temp_5": "17.0",
  "sunday_temp_6": "21.0",
  "sunday_temp_7": "17.0",
  "sunday_temp_8": "21.0",
  "sunday_temp_9": "17.0",
  "system_mode": "auto",
  "thursday_hour_1": "06",
  "thursday_hour_2": "09",
  "thursday_hour_3": "17",
  "thursday_hour_4": "23",
  "thursday_hour_5": "24",
  "thursday_hour_6": "24",
  "thursday_hour_7": "24",
  "thursday_hour_8": "24",
  "thursday_minute_1": "00",
  "thursday_minute_2": "00",
  "thursday_minute_3": "00",
  "thursday_minute_4": "00",
  "thursday_minute_5": "00",
  "thursday_minute_6": "00",
  "thursday_minute_7": "00",
  "thursday_minute_8": "00",
  "thursday_temp_1": "17.0",
  "thursday_temp_2": "21.0",
  "thursday_temp_3": "17.0",
  "thursday_temp_4": "21.0",
  "thursday_temp_5": "17.0",
  "thursday_temp_6": "21.0",
  "thursday_temp_7": "17.0",
  "thursday_temp_8": "21.0",
  "thursday_temp_9": "17.0",
  "tuesday_hour_1": "06",
  "tuesday_hour_2": "09",
  "tuesday_hour_3": "17",
  "tuesday_hour_4": "23",
  "tuesday_hour_5": "24",
  "tuesday_hour_6": "24",
  "tuesday_hour_7": "24",
  "tuesday_hour_8": "24",
  "tuesday_minute_1": "00",
  "tuesday_minute_2": "00",
  "tuesday_minute_3": "00",
  "tuesday_minute_4": "00",
  "tuesday_minute_5": "00",
  "tuesday_minute_6": "00",
  "tuesday_minute_7": "00",
  "tuesday_minute_8": "00",
  "tuesday_temp_1": "17.0",
  "tuesday_temp_2": "21.0",
  "tuesday_temp_3": "17.0",
  "tuesday_temp_4": "21.0",
  "tuesday_temp_5": "17.0",
  "tuesday_temp_6": "21.0",
  "tuesday_temp_7": "17.0",
  "tuesday_temp_8": "21.0",
  "tuesday_temp_9": "17.0",
  "voltage": 920,
  "wednesday_hour_1": "06",
  "wednesday_hour_2": "09",
  "wednesday_hour_3": "17",
  "wednesday_hour_4": "23",
  "wednesday_hour_5": "24",
  "wednesday_hour_6": "24",
  "wednesday_hour_7": "24",
  "wednesday_hour_8": "24",
  "wednesday_minute_1": "00",
  "wednesday_minute_2": "00",
  "wednesday_minute_3": "00",
  "wednesday_minute_4": "00",
  "wednesday_minute_5": "00",
  "wednesday_minute_6": "00",
  "wednesday_minute_7": "00",
  "wednesday_minute_8": "00",
  "wednesday_temp_1": "17.0",
  "wednesday_temp_2": "21.0",
  "wednesday_temp_3": "17.0",
  "wednesday_temp_4": "21.0",
  "wednesday_temp_5": "17.0",
  "wednesday_temp_6": "21.0",
  "wednesday_temp_7": "17.0",
  "wednesday_temp_8": "21.0",
  "wednesday_temp_9": "17.0"
}
`

Any suggestions?

**Comment by louisnichols** [comment]
OK. So I just noticed that the new schedule also appears on the "State" tab of the device page. Basically, exactly as in MQTT. Isn't there a way to upload this to the TRV?

**Comment by louisnichols** [comment]
Moving on on this voyage of discovery. :) It dawned on me at some point to check the thermostat itself. And the schedule for Monday is actually stored, just as I sent it via MQTT. So... yay.

But still, shouldn't I see these updates on the Exposes tab of the device? And in the dashboard?

**Comment by louisnichols** [comment]
Anyone? Anything? :)

**Comment by tovadas** [comment]
See [New Device Issue #19806](https://github.com/Koenkk/zigbee2mqtt/issues/19806).
